### PR TITLE
Introduce pkg/grpcutil

### PR DESCRIPTION
**Pull request by jzelinskie**
This change refactors all of the gRPC code used in the v3 package to be generic and reusable.

### CODE REVIEW

File: pkg/grpcutil/gateway.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+	} else {
+		dialOpts = append(dialOpts, grpc.WithInsecure())
+	}
+
+	srvmux := runtime.NewServeMux(
+		// Include empty fields with default values.
+		runtime.WithMarshalerOption(runtime.MIMEWildcard, &runtime.JSONPb{EmitDefaults: true}),
+	)
+
+	conn, err := grpc.DialContext(context.TODO(), addr, dialOpts...)
```
Comment (KeyboardNerd) [inline pkg/grpcutil/gateway.go]: It was using context.Background() in the old code. Should we still use that?

File: pkg/grpcutil/gateway.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+// NewGatewayHandler creates a new http.Handler with the provided gRPC Services
+// registered.
+func NewGatewayHandler(addr string, tlsConfig *tls.Config, funcs []RegisterServiceHandlerFunc) (http.Handler, error) {
+	// Configure the right DialOptions the for TLS configuration.
+	var dialOpts []grpc.DialOption
+	if tlsConfig != nil {
+		var gwTLSConfig *tls.Config
+		gwTLSConfig = tlsConfig.Clone()
+		gwTLSConfig.InsecureSkipVerify = true
```
Comment (KeyboardNerd) [inline pkg/grpcutil/gateway.go]: Can we add a comment on why we use InsecureSkipVerify here?

File: pkg/grpcutil/gateway.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+		dialOpts = append(dialOpts, grpc.WithTransportCredentials(credentials.NewTLS(gwTLSConfig)))
+	} else {
+		dialOpts = append(dialOpts, grpc.WithInsecure())
+	}
+
+	srvmux := runtime.NewServeMux(
+		// Include empty fields with default values.
+		runtime.WithMarshalerOption(runtime.MIMEWildcard, &runtime.JSONPb{EmitDefaults: true}),
```
Comment (KeyboardNerd) [inline pkg/grpcutil/gateway.go]: Do you remember why do we want to include the empty fields?

**Review by KeyboardNerd — COMMENTED** [review]
Some comments on gateway.go
We should also add tests on server connectivity in the future.

File: pkg/grpcutil/gateway.go (hunk 1 above)
Comment (jzelinskie) [reply pkg/grpcutil/gateway.go]: We should use `TODO()` until the code has been entirely refactored to support context from the start of the application. `Background()` is how you'd initialize a context if you were doing so from the proper location, but in this case, we'll have to handle that another time.

Comment (KeyboardNerd) [reply pkg/grpcutil/gateway.go]: Sure

File: pkg/grpcutil/gateway.go (hunk 3 above)
Comment (jzelinskie) [reply pkg/grpcutil/gateway.go]: Nope

Comment (KeyboardNerd) [reply pkg/grpcutil/gateway.go]: Yeah, just get rid of it, I don't think there will be any issues.

**Review by KeyboardNerd — APPROVED** [review]
LGTM

**Pull request merged**